pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -121,6 +149,10 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	params := config.Spec.Strategy.RollingParams
 	updateAcceptor := s.getUpdateAcceptor(time.Duration(*params.TimeoutSeconds)*time.Second, config.Spec.MinReadySeconds)
 
+	// Record all warnings
+	defer s.recordControllerWarnings(from)
```

Comment:
Moving the defer statements you have here after the if clause below fixes the duplicate message
```
$ oc deploy dc/ruby-ex --latest --follow
Started deployment #3
--> Scaling ruby-ex-3 to 3
-->  FailedCreate: ruby-ex-3 Error creating: pods "ruby-ex-3-" is forbidden: Exceeded quota: quota, requested: pods=1, used: pods=2, limited: pods=2
error: couldn't scale ruby-ex-3 to 3: timed out waiting for "ruby-ex-3" to be synced
```
That's because for initial rolling deployments, the defer statements are executed twice, once here and once in the deployer acceptor we are calling for delegating to recreate.